kernel/api/src/main/java/org/sakaiproject/exception/ZippedFilenumberException.java

```diff
@@ -0,0 +1,30 @@
+/**********************************************************************************
+ * Copyright (c) 2019 Apereo Foundation
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *       http://opensource.org/licenses/ecl2
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **********************************************************************************/
+
+package org.sakaiproject.exception;
+
+/**
+ * ZippedFilenumberException is thrown whenever a ZIP file contains more files than the maximum allowed.
+ */
+public class ZippedFilenumberException extends SakaiException
+{
+
+    /**
+     * @param id The resource that contains more files than the allowed.
+     */
+    public ZippedFilenumberException(String id) {
```

Comment:
ZipFileNumberException seems ideal

Should this exception record how many files were attempted and what the maximum is?

Reply:
Ok, changed the name